### ISSUE TITLE
[Question] Reading master file table

### ISSUE THREAD

**Issue by Solido** [halildurmus/win32#244]
Hi Tim,

Would it be possible to access the master file table from dart using your win 32 binding ?

Thx !

**Comment by timsneath** [comment]
You mean, the NTFS MFT? Or do you mean something else? Curious and fascinated and slightly alarmed by the use case...? :) 

If you can point me at the Win32 API you'd like wrapped, I'll happily take a look. The docs are here: https://docs.microsoft.com/en-us/windows/win32/fileio/about-volume-management

**Comment by Solido** [comment]
Yes I am looking at c:\$mft

Well I got contacted by a specific agent that will not allow me to disclose how a specific tool implement the doctrine :)
at least in public but I'll send some details in your twitter DM if you agree :D

To my very limited knowledge it has to do with DeviceIoControl

**Comment by timsneath** [comment]
Not quite what you want, but I've got _an_ example of `DeviceIoControl` checked into main now:
https://github.com/timsneath/win32/blob/main/example/diskinfo.dart

I spent some time spelunking, but it gets pretty hairy beyond this point to parse the MFT. I spent a little while reading around, but it's not an interface that is designed for easy perusal. All the IOCTL codes are checked into the package, but you probably have some work on your hands if you want to go down this route. 

`GetFileSizeEx`, on the other hand, is much easier :)
